lib/Controller/LocationsController.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Varun Patil <radialapps@gmail.com>
+ * @author Varun Patil <radialapps@gmail.com>, Raymond Huang <raymond860909@gmail.com>
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Memories\Controller;
+
+use OCA\Memories\Db\TimelineRoot;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+
+class LocationsController extends ApiBase
+{
+    use FoldersTrait;
+
+    /**
+     * @NoAdminRequired
+     *
+     * @PublicPage
+     */
+    public function daysWithBounds(string $minLat, string $maxLat, string $minLng, string $maxLng): JSONResponse
```

Comment:
This code is duplicating the Days code. It should programmed as a transform instead.

Reply:
Could you explain more about programmed as a transform? Do you mean polymorphism, or is this a language feature specific to PHP?